feat: warn users who are falling behind reprovides (#9886)
ipfs/kubo: docs/changelogs/v0.21.md

- [Overview](#overview)
- [🔦 Highlights](#-highlights)
  - [Accelerated DHT Client is no longer experimental](#--empty-repo-is-now-the-default)
- [📝 Changelog](#-changelog)
- [👨‍👩‍👧‍👦 Contributors](#-contributors)

### Overview

### 🔦 Highlights

#### Accelerated DHT Client is no longer experimental

> The accelerated DHT client is now the main recomended solution for peoples who

Reviewer: ```suggestion
The [accelerated DHT client](docs/config.md#routingaccelerateddhtclient) is now the main recommended solution for users who
```

> are hosting lots of data.

Reviewer: ```suggestion
are hosting lots of data.  By trading some upfront DHT caching and increased memory usage, one gets provider latency improvements of Xx.  See [the docs](docs/config.md#routingaccelerateddhtclient) for more info.
```

Author: I'll remove
> one gets provider latency improvements of Xx

it's not the reason we are telling people to use it, it's mainly throughput

> We moved the `Experimental.AcceleratedDHTClient` flag in `Routing.AcceleratedDHTClient`.

Reviewer: ```suggestion
The `Experimental.AcceleratedDHTClient` flag moved to `[Routing.AcceleratedDHTClient](docs/config.md#routingaccelerateddhtclient)`.  A repo migration has been added.
```

> A new tracker will try to estimate the providing speed and warn users that they
> should be using AcceleratedDHTClient because they are falling behind.

Reviewer: ```suggestion
A new tracker estimates the providing speed and warns users if they
should be using AcceleratedDHTClient because they are falling behind.
```
